homeassistant/components/econet/__init__.py

```diff
@@ -1 +1,183 @@
-"""The econet component."""
+"""Support for EcoNet products."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyeconet import EcoNetApiInterface
+from pyeconet.equipment import EquipmentType
+from pyeconet.errors import InvalidCredentialsError, PyeconetError
+
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, TEMP_FAHRENHEIT
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import API_CLIENT, DOMAIN, EQUIPMENT
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ON_VACATION = "on_vacation"
+ATTR_IN_USE = "in_use"
+ATTR_WIFI_SIGNAL = "wifi_signal_strength"
+ATTR_ALERT_COUNT = "alert_count"
+PLATFORMS = ["water_heater"]
+
+INTERVAL = timedelta(minutes=60)
+
+
+async def async_setup(hass, config):
+    """Set up the EcoNet component."""
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][API_CLIENT] = {}
+    hass.data[DOMAIN][EQUIPMENT] = {}
+
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+    _LOGGER.error(conf)
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_EMAIL: conf[CONF_EMAIL],
+                CONF_PASSWORD: conf[CONF_PASSWORD],
+            },
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up EcoNet as config entry."""
+    entry_updates = {}
+    if entry_updates:
+        hass.config_entries.async_update_entry(config_entry, **entry_updates)
+
+    email = config_entry.data.get(CONF_EMAIL)
+    password = config_entry.data.get(CONF_PASSWORD)
+
+    try:
+        api = await EcoNetApiInterface.login(email, password=password)
+    except InvalidCredentialsError:
+        _LOGGER.error("Invalid credentials provided")
+        return False
+    except PyeconetError as err:
+        _LOGGER.error("Config entry failed: %s", err)
+        raise ConfigEntryNotReady from err
+
+    equipment = await api.get_equipment_by_type([EquipmentType.WATER_HEATER])
+    hass.data[DOMAIN][API_CLIENT][config_entry.entry_id] = api
+    hass.data[DOMAIN][EQUIPMENT][config_entry.entry_id] = equipment
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
+
+    api.subscribe()
+
+    async def resubscribe(now):
+        """Resubscribe to the MQTT updates."""
+        api.unsubscribe()
```

Comment:
This is a blocking call:
https://github.com/w1ll1am23/pyeconet/blob/d9b12a620a30e3b507e872ad104217cf067ccf90/src/pyeconet/api.py#L104-L105

https://github.com/eclipse/paho.mqtt.python/blob/225ab3757f6818ba85eb80564948d1c787190cba/src/paho/mqtt/client.py#L1824-L1827

Reply:
fixed